pkg/packagekit/updates.jsx

```diff
@@ -73,6 +74,25 @@ function init() {
         [PK.Enum.STATUS_CLEANUP]: _("Set up"),
         [PK.Enum.STATUS_SIGCHECK]: _("Verified"),
     };
+
+    // This list is inspired by https://access.redhat.com/solutions/27943
+    // It's done the same way by yum util needs-restarting
+    REBOOT_PKGS = [
+        "kernel",
+        "kernel-rt",
+        "glibc",
+        "linux-firmware",
+        "systemd",
+        "udev",
+        "openssl-libs",
+        "gnutls",
+        "dbus",
+        "dbus-daemon",
+        "gvfsd-metadata",
+        "at-spi-bus-launcher",
+        "gconfd-2",
+        "cockpit-testing-package", // This package is here only for Cockpit's testing purposes
+    ];
```

Comment:
This list is too large -- e. g. you certainly don't need a reboot for udev. At least on Debian/Ubuntu/Fedora you don't need a reboot for systemd either, as the package postinst does `systemctl daemon-reexec`. It's also incomplete -- if dbus-daemon requires a reboot, then dbus-broker surely needs one either.

These package names also are RHEL/Fedora specific, and won't match on a Debian system.

Quite frankly, I find this a terrible hack, and I don't want it. This list will get out of date very quickly, and nothing will visibly break when it does. Let's please do this properly with tracer (Fedora/RHEL) and needrestart (Debian/Ubuntu).

Reply:
Tracer itself doesn't provide an API to just check if a reboot is needed. You have to run the whole (slow) thing and then it tells you apart from other things that reboot is needed, which is for this case not really a good solution. needs-restarting and needrestart can do this, so another dependency is needed.

needs-restarting for Rhel/Fedora dna needrestart for Debian and Ubuntu works fine. But it won't tell us if a reboot may be needed before the update happens (which is a nice functionality of this PR). So I would perhaps keep this list to warn the user before the update that reboot MAY be needed once they update, and then after the update use above mentioned tools to tell if a reboot is REQUIRED. WDYT?

Reply:
Also, needs-restarting uses this list too (https://github.com/rpm-software-management/yum-utils/blob/master/needs-restarting.py#L53), but yeah at least we wouldn't have to update it ourself, and it seems that needrestart just checks kernel version.

Reply:
As discussed yesterday, we eventually want to run the full tracer anyway, to get the list of services to be restarted. *That* is the real benefit  of this work, to *avoid* a reboot. :grin: So let's just take the plunge and call it, instead of coming up with complicated temporary workarounds.

I'm open to discuss the "may be required" part -- but honestly, I don't see too much value in it. If you apply updates and get a notification then, that's not any worse than not doing the update at all. The old kernel will still run either way, and at least you get other system components updated.